test/exportapi.test.js

```diff
@@ -0,0 +1,180 @@
+/*global describe, beforeEach, it */
+'use strict';
+var path = require('path');
+var helpers = require('yeoman-generator').test;
+var SANDBOX =  path.resolve(__dirname, 'sandbox');
+var fs = require('fs');
+var expect = require('chai').expect;
+var wsModels = require('loopback-workspace').models;
+var common = require('./common');
+var yaml = require('js-yaml');
+
+describe('loopback:model generator', function() {
+  beforeEach(common.resetWorkspace);
+
+  beforeEach(function createSandbox(done) {
+    helpers.testDirectory(SANDBOX, done);
+  });
+
+  beforeEach(function createProject(done) {
+    common.createDummyProject(SANDBOX, 'test-app', done);
+  });
+
+  beforeEach(function addRestDataSource(done) {
+    wsModels.DataSourceDefinition.create({
+      name: 'rest',
+      connector: 'rest',
+      facetName: 'server'
+    }, done);
+  });
+
+  it('creates common/models/{name}.json', function(done) {
```

Comment:
`loopback:export-api-def creates common/models/{name}.json` - that does not sound right. If the tests for `export-api-def` need more models to be created, then create then in `beforeEach`  hook(s), not in a test case.

```js
beforeEach(function createProductModel(done) {
  var modelGen = givenModelGenerator();
  // etc.
});
```

Since we are not testing the model generator, it's better to use `wsModels.ModelDefinition` API directly.